tools/src/versioning/android/versionVendoredModules.ts

```diff
@@ -0,0 +1,227 @@
+    'library_jni',
+    'release',
+    'jni'
+  );
+  const libFiles = await glob('**/*.so', {
+    cwd: buildLibDir,
+  });
+  assert(libFiles.length >= 0);
+  await Promise.all(
+    libFiles.map(async (file) => {
+      const srcPath = path.join(buildLibDir, file);
+      const archName = path.basename(path.dirname(file));
+      const dstPath = path.join(jniLibDir, archName, path.basename(file));
+      await fs.ensureDir(path.dirname(dstPath));
+      await fs.copy(srcPath, dstPath);
+    })
+  );
+
+  // Truncate CMakeLists.txt and not to build this cxx module when building versioned Expo Go
+  await fs.writeFile(cmakeFile, '');
+  await fs.remove(path.join(moduleRootDir, 'build'));
+}
+
+/**
+ * Gets the library name of each vendored module in a specific directory.
+ */
+async function getVendoredModuleNamesAsync(directory: string): Promise<string[]> {
+  const vendoredGradlePaths = await glob(`**/build.gradle`, {
+    cwd: directory,
+    nodir: true,
+    realpath: true,
+  });
+
+  const gradlePattern = new RegExp(`${directory}/(.*)/android/build.gradle`, 'i');
+
+  return vendoredGradlePaths.reduce((result, gradlePath) => {
+    const moduleName = gradlePath.match(gradlePattern);
+    if (moduleName) {
+      result.push(moduleName[1]);
+    }
+    return result;
+  }, [] as string[]);
+}
+
+/**
+ * Removes the directory with vendored modules for given SDK number.
+ */
+export async function removeVersionedVendoredModulesAsync(sdkNumber: number): Promise<void> {
+  const versionedDir = vendoredDirectoryForSDK(sdkNumber);
+  await fs.remove(versionedDir);
+}
+
+/**
+ * Generates base transforms to apply for all vendored modules.
+ */
+function baseTransformsFactory(prefix: string): Required<FileTransforms> {
+  return {
+    path: [
+      {
+        // For package renaming, src/main/java/* -> src/main/java/abiN/*
+        find: /\/java\//,
+        replaceWith: `/java/${prefix}/`,
+      },
+    ],
+    content: [
+      {
+        paths: '*.{java,kt}',
+        find: /(^package\s+)([\w.]+;?)/m,
+        replaceWith: `$1${prefix}.$2`,
+      },
+      {
+        paths: '*.{java,kt}',
+        find: /(\bcom\.facebook\.(catalyst|csslayout|fbreact|hermes|perftest|quicklog|react|systrace|yoga|debug)\b)/g,
+        replaceWith: `${prefix}.$1`,
+      },
+      {
+        paths: '*.{java,kt}',
+        find: /\b((System|SoLoader)\.loadLibrary\("[^"]*)("\);?)/g,
+        replaceWith: `$1_${prefix}$3`,
+      },
+      {
+        paths: '*.{h,cpp}',
+        find: /(\bkJavaDescriptor\s*=\s*\n?\s*"L)/gm,
+        replaceWith: `$1${prefix}/`,
+      },
+      {
+        paths: 'build.gradle',
+        find: `implementation 'com.facebook.react:react-native:+'`,
+        replaceWith:
+          `implementation 'host.exp:reactandroid-${prefix}:1.0.0'` +
+          '\n' +
+          // Adding some compile time common dependencies where the versioned react-native AAR doesn't expose
+          `    compileOnly 'com.facebook.fbjni:fbjni:0.2.2'\n` +
```

Comment:
hardcoding versions in versioning script? wouldn't this require manual changes every time we update rn?